packages/babel-helper-module-transforms/src/rewrite-live-references.ts

```diff
@@ -355,31 +355,33 @@ const rewriteReferencesVisitor: Visitor<RewriteReferencesVisitorState> = {
   ) {
     const { scope, node } = path;
     const { left } = node;
-    const { exported, scope: programScope } = this;
+    const { exported, imported, scope: programScope } = this;
 
     if (!t.isVariableDeclaration(left)) {
-      let didTransform = false;
-      const bodyPath = path.get("body");
-      const loopBodyScope = bodyPath.scope;
+      let didTransformExport = false,
+        importConstViolationName;
+      const loopBodyScope = path.get("body").scope;
       for (const name of Object.keys(t.getOuterBindingIdentifiers(left))) {
-        if (
-          exported.get(name) &&
-          programScope.getBinding(name) === scope.getBinding(name)
-        ) {
-          didTransform = true;
-          if (loopBodyScope.hasOwnBinding(name)) {
-            loopBodyScope.rename(name);
+        if (programScope.getBinding(name) === scope.getBinding(name)) {
+          if (exported.get(name)) {
+            didTransformExport = true;
+            if (loopBodyScope.hasOwnBinding(name)) {
+              loopBodyScope.rename(name);
+            }
+          }
+          if (imported.get(name) && !importConstViolationName) {
+            importConstViolationName = name;
```

Comment:
Should we record all const violation names? e.g.

```js
import { x, y } from 'foo';
for ( [x, y] of [] );
```

Reply:
It's enough to report the first one, since we only inject the error for the first one (the others are not observable anyway).

Reply:
I think @nicolo-ribaudo is correct. Only the first is necessary as that will throw an error before the 2nd is evaluated.